client/src/app/appCtrl.js

```diff
@@ -46,36 +60,127 @@ angular.module('app').controller('appCtrl', ['$scope', '$http', '$timeout', 'app
     slidingWindow = false;
     $scope.$broadcast("fileUploadChange");
     $scope.view.loading = true;
-    // we do a quick test here to see if the server supports the Range header.
-    // If so, we try to stream. If not, we try to download.
-    $http.head($scope.view.filePath,{'headers' : {'Range' : 'bytes=0-32'}}).then(function(response){
-      if(response.status === 206) {
-        // now we check to see how big the file is
-        $http.head($scope.view.filePath).then(function(response){
-          var contentLength = response.headers('Content-Length');
-          if (contentLength > appConfig.MAX_FILE_SIZE) {
-            slidingWindow = true;
-            $scope.view.windowing.show = true;
-          }
-          streamRemoteFile($scope.view.filePath);
-        });
-      } else {
-        downloadFile($scope.view.filePath);
-      }
-    }, function() {
-      downloadFile($scope.view.filePath);
-    });
+    $scope.view.file.size = getFileSize($scope.view.filePath);
+    console.log("File size "+$scope.view.file.size);
+    if ($scope.canDownload() && $scope.view.file.size > appConfig.MAX_FILE_SIZE) { 
+      console.log("streamig...");
+      downloadFile($scope.view.filePath, "streamRemote");
+    } else {
+      downloadFile($scope.view.filePath, "download"); // fallback, or small file
+    }  
+  };
+
+  // handle downloading (or Browsing) a local file
+  $scope.getLocalFile = function() {
+    // update file size for local files
+    $scope.view.file.size = getFileSize($scope.view.file.file);
+    console.log("File size "+$scope.view.file.size);
+    if ($scope.view.file.size > appConfig.MAX_FILE_SIZE) {
+      slidingWindow = true;
+      $scope.view.windowing.show = true;
+    }
+    $scope.view.loading = true;
+    downloadFile($scope.view.file.file, "streamLocal");
+  };
+
+  // main "load" function that supports both URL/local file
+  // takes care of monitoring/streaming-data plots: if appConfig.POLLING_INTERVAL > 0 keep polling the file,sleeping
+  $scope.loadFile = function(event) {
```

Comment:
* I needed a single "load" function for the use in monitoring/polling. 
(about the UI, I thought we can get rid of one of the buttons, like "if something in text_entry && load" -> assume URL, "if text empty && load" -> assume local file browse; but that is not a priority, just small interface cleanup)